src/components/node-list/node-list.js

```diff
@@ -1,44 +1,64 @@
-import React from 'react';
+import React, { useContext, useEffect, useState } from 'react';
+import debounce from 'lodash/debounce';
 import classnames from 'classnames';
 import { Scrollbars } from 'react-custom-scrollbars-2';
 import SearchList from '../search-list';
-import NodeListGroups from './node-list-groups';
+import Filters from '../filters/filters';
 import NodeListTree from './node-list-tree';
 import SplitPanel from '../split-panel';
+import { FiltersContext } from './utils/filters-context';
+import { NodeListContext } from './utils/node-list-context';
+import { getModularPipelinesSearchResult } from '../../selectors/modular-pipelines';
 
 import './styles/node-list.scss';
 
 /**
  * Scrollable list of toggleable items, with search & filter functionality
  */
-const NodeList = ({
-  faded,
-  items,
-  modularPipelinesTree,
-  modularPipelinesSearchResult,
-  groups,
-  searchValue,
-  getGroupState,
-  onUpdateSearchValue,
-  onGroupToggleChanged,
-  onItemClick,
-  onItemMouseEnter,
-  onItemMouseLeave,
-  onItemChange,
-  onModularPipelineToggleExpanded,
-  focusMode,
-  disabledModularPipeline,
-  onResetFilter,
-  isResetFilterActive,
-}) => {
+const NodeList = ({ faded }) => {
+  const [searchValue, updateSearchValue] = useState('');
+
+  const {
+    groupCollapsed,
+    groups,
+    isResetFilterActive,
+    items,
+    handleGroupToggleChanged,
+    handleResetFilter,
+    handleToggleGroupCollapsed,
+    handleFiltersRowClicked,
+  } = useContext(FiltersContext);
+
+  const {
+    modularPipelinesTree,
+    handleModularPipelineToggleExpanded,
+    handleNodeListRowClicked,
+    handleNodeListRowChanged,
+    handleItemMouseEnter,
+    handleItemMouseLeave,
+    handleToggleHoveredFocusMode,
+    focusMode,
+    disabledModularPipeline,
+    handleKeyDown,
+  } = useContext(NodeListContext);
+
+  const modularPipelinesSearchResult = searchValue
```

Comment:
this can be moved to node-list-tree

Reply:
yup good point 👍

Reply:
actually now looking at the code, i think it makes more sense to do this at the node-list, then just pass down the `modularPipelinesSearchResult` to the `node-list-tree` as a prop